File: src/main/java/com/mateusz/rest/webservices/restful_web_services/users/PostNotFoundException.java
package com.mateusz.rest.webservices.restful_web_services.users;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(code = HttpStatus.NOT_FOUND)
public class PostNotFoundException extends RuntimeException {// avoid checked exception
    public PostNotFoundException(String message) {
        super(message);// pass msg to runtime exception
    }
}

File: src/main/kotlin/com/mateusz/rest/webservices/restful_web_services/spring_boot_social_media_restful_api/users/PostNotFoundException.kt
package com.mateusz.rest.webservices.restful_web_services.spring_boot_social_media_restful_api.users

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus


@ResponseStatus(code = HttpStatus.NOT_FOUND)
class PostNotFoundException  // avoid checked exception
    (message: String?) : RuntimeException(message)
